osg-comanage-project-usermap.py

```diff
@@ -87,8 +94,21 @@ def mkrequest(target, **kw):
 
 def call_api(target, **kw):
     req = mkrequest(target, **kw)
-    resp = urllib.request.urlopen(req)
-    payload = resp.read()
+    trying = True
+    currentTimeout = options.minTimeout
+    while trying:
+        try:
+            resp = urllib.request.urlopen(req, timeout=currentTimeout)
+            payload = resp.read()
+            trying = False
+        except urllib.error.URLError as exception:
+            if (currentTimeout < options.maxTimeout):
+                currentTimeout *= TIMEOUTMULTIPLE
+            else:
+                print("Exception raised after maximum timeout " + str(options.maxTimeout) + " seconds reached. Exception reason: " + str(exception.reason) + ".\n Request: " + str(req.full_url))
+                trying = False
+                quit()
```

Comment:
You should use `sys.exit()` instead. Also, no need to set `trying` if you raise the SystemExit exception via your exit function. Also you can simplify all this to one line by adding your error message as the arg to `sys.exit()`.

Reply:
Replaced quit() with sys.exit() with the f-string passed in commit 80e513c